pandas/core/indexes/multi.py

```diff
@@ -2178,6 +2209,47 @@ def append(self, other):
         except (TypeError, IndexError):
             return Index(new_tuples)
 
+    def get_index_data(self, data_index, column_name, other, search_self=False):
+
+        # Returns original data if the data_index input has data for this column name
+        if column_name in data_index.names:
+            Index_position = data_index.names.index(column_name)
+            data = data_index._get_level_values(Index_position)
+            return data
+
```

Comment:
If the MultIndex being checked has the column name, it would just return the original data that is in the MultiIndex